File: app/src/main/java/com/zibete/proyecto1/SlideProfileActivity.java
package com.zibete.proyecto1;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.ValueEventListener;
import com.zibete.proyecto1.Adapters.SliderProfileAdapter;
import com.zibete.proyecto1.POJOS.Users;

import java.util.ArrayList;
import java.util.Objects;

import static com.zibete.proyecto1.Constants.chatWith;
import static com.zibete.proyecto1.MainActivity.ref_datos;

public class SlideProfileActivity extends AppCompatActivity {

    FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
    ProgressBar progressbarImage;

    ArrayList <Users> userList;
    ViewPager viewPager;

    public SlideProfileActivity(){
        //Constructor vacío
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.slide_activity);




        progressbarImage = findViewById(R.id.progressbarImage);
        progressbarImage.setVisibility(View.GONE);
        Toolbar toolbar = findViewById(R.id.toolbar2);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        userList = (ArrayList<Users>) getIntent().getExtras().getSerializable("userList");

        final int position = getIntent().getExtras().getInt("position");
        final int rotation = getIntent().getExtras().getInt("rotation");




        LinearLayout linearSlide = findViewById(R.id.linearSlide);

        viewPager = new ViewPager(this);
        viewPager.setId(View.generateViewId());


        viewPager.setAdapter(new SliderProfileAdapter(this, userList, rotation));



        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);

        //viewPager.setRotationY(180);
        //viewPager.setLayoutDirection(View.LAYOUT_DIRECTION_RTL);


        //linearSlide.setRotationY(180);


        viewPager.setLayoutParams(params);



        linearSlide.addView(viewPager, viewPager.getLayoutParams());
        viewPager.setCurrentItem(position);



    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();

    }


    @Override
    public boolean onCreateOptionsMenu(final Menu menu) {
        getMenuInflater().inflate(R.menu.menu_chat_activity, menu);


        return super.onCreateOptionsMenu(menu);
    }




    @Override
    public boolean onPrepareOptionsMenu(final Menu menu) {
        super.onPrepareOptionsMenu(menu);

        final MenuItem action_silent = menu.findItem(R.id.action_silent);
        final MenuItem action_notif = menu.findItem(R.id.action_notif);
        final MenuItem action_bloq = menu.findItem(R.id.action_bloq);
        final MenuItem action_desbloq = menu.findItem(R.id.action_desbloq);
        final MenuItem action_delete = menu.findItem(R.id.action_delete);


        action_delete.setVisible(true);

        Users users = userList.get(viewPager.getCurrentItem());
        //Para el menu1
        ref_datos.child(user.getUid()).child(chatWith).child(users.getID()).addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {

                if (dataSnapshot.exists()){

                    String state = dataSnapshot.child("estado").getValue(String.class);

                    if(Objects.equals(state, "silent")){
                        action_silent.setVisible(false);
                        action_notif.setVisible(true);
                        action_desbloq.setVisible(false);
                        action_bloq.setVisible(true);
                    }
                    if(Objects.equals(state, chatWith)){
                        action_silent.setVisible(true);
                        action_notif.setVisible(false);
                        action_desbloq.setVisible(false);
                        action_bloq.setVisible(true);
                    }
                    if(Objects.equals(state, "bloq")){
                        action_silent.setVisible(false);
                        action_notif.setVisible(false);
                        action_desbloq.setVisible(true);
                        action_bloq.setVisible(false);
                    }
                    if(Objects.equals(state, "delete")){
                        action_silent.setVisible(true);
                        action_notif.setVisible(false);
                        action_desbloq.setVisible(false);
                        action_bloq.setVisible(true);
                    }

                }else{

                    action_silent.setVisible(true);
                    action_notif.setVisible(false);
                    action_desbloq.setVisible(false);
                    action_bloq.setVisible(true);

                }

            }
            @Override
            public void onCancelled(@NonNull DatabaseError error) {
            }
        });


        return true;
    }





/*
    @Override
    public boolean onContextItemSelected(MenuItem item) {

        if(item.getGroupId() == 1) {

            final Users users = userList.get(item.getOrder());
            String id_user = users.getID();
            String name_user = users.getNombre();



            View view = findViewById(android.R.id.content);

            if (item.getItemId() == android.R.id.home) {
                super.onBackPressed();
                return true;
            }





            switch (item.getItemId()) {
                case R.id.action_silent: //Silenciar notificaciones
                    new Constants().Silenciar(id_user);
                    Toast.makeText(this, "Notificaciones desactivadas", Toast.LENGTH_SHORT).show();

                    break;

                case R.id.action_notif:

                    new Constants().Silenciar(id_user);
                    Toast.makeText(this, "Notificaciones activadas", Toast.LENGTH_SHORT).show();

                    break;

                case R.id.action_bloq: //Bloquear


                    new Constants().Bloquear(this, id_user, name_user, view);

                    break;

                case R.id.action_desbloq: //DESBloquear

                    new Constants().desBloquear(this, id_user, name_user, view);

                    break;


                case R.id.action_delete: //Eliminar

                    View vista = getLayoutInflater().inflate(R.layout.dialog_delete,null);
                    new Constants().Eliminar(this, id_user, name_user, vista);

                    break;
            }
        }
        return true;
    }

*/












    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        Users users = userList.get(viewPager.getCurrentItem());
        View view = findViewById(android.R.id.content);
        int id = item.getItemId();

        if (id == android.R.id.home) {
            super.onBackPressed();

        } else if (id == R.id.action_silent) { // Silenciar notificaciones
            new Constants().Silent(users.getNombre(), users.getID(), chatWith);
            Toast.makeText(this, "Notificaciones desactivadas", Toast.LENGTH_SHORT).show();

        } else if (id == R.id.action_notif) {
            new Constants().Silent(users.getNombre(), users.getID(), chatWith);
            Toast.makeText(this, "Notificaciones activadas", Toast.LENGTH_SHORT).show();

        } else if (id == R.id.action_bloq) { // Bloquear
            new Constants().Block(this, users.getNombre(), users.getID(), view, chatWith);

        } else if (id == R.id.action_desbloq) { // Desbloquear
            new Constants().desBloquear(this, users.getID(), users.getNombre(), view, chatWith);

        } else if (id == R.id.action_delete) { // Eliminar
            new Constants().DeleteChat(this, users.getID(), users.getNombre(), view, chatWith);
        }

        return super.onOptionsItemSelected(item);
    }















    @Override
    protected void onPause() {
        super.onPause();
        new Constants().StateOffLine(getApplicationContext(),user.getUid());
    }

    @Override
    protected void onResume() {
        super.onResume();
        new Constants().StateOnLine(getApplicationContext(), user.getUid());
    }

}

File: app/src/main/java/com/zibete/proyecto1/SlideProfileActivity.kt
package com.zibete.proyecto1

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ProgressBar
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.lifecycleScope
import androidx.viewpager2.widget.ViewPager2
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener
import com.zibete.proyecto1.adapters.SliderProfileAdapter
import com.zibete.proyecto1.data.UserRepository
import com.zibete.proyecto1.di.firebase.FirebaseRefsContainer
import com.zibete.proyecto1.model.Users
import com.zibete.proyecto1.ui.base.BaseToolbarActivity
import com.zibete.proyecto1.ui.constants.Constants
import com.zibete.proyecto1.utils.ProfileUiBinder
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch
import javax.inject.Inject

@AndroidEntryPoint
class SlideProfileActivity : BaseToolbarActivity() {

    @Inject lateinit var profileUiBinder: ProfileUiBinder
    @Inject lateinit var userRepository: UserRepository
    @Inject lateinit var firebaseRefsContainer: FirebaseRefsContainer


    private val user = userRepository.user

    private lateinit var progressbarImage: ProgressBar
    private lateinit var userList: MutableList<Users>
    private lateinit var viewPager: ViewPager2

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.slide_activity)

        progressbarImage = findViewById(R.id.progressbarImage)
        progressbarImage.visibility = android.view.View.GONE

        val toolbar = findViewById<Toolbar>(R.id.toolbar2)
        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            title = ""
            setDisplayHomeAsUpEnabled(true)
        }
        toolbar.setNavigationOnClickListener { onBackPressedDispatcher.onBackPressed() }

        @Suppress("UNCHECKED_CAST")
        userList = (intent.extras?.getSerializable("userList") as? MutableList<Users>) ?: mutableListOf()
        val position = intent.extras?.getInt("position") ?: 0
        val rotation = intent.extras?.getInt("rotation") ?: 0

        viewPager = findViewById(R.id.viewPager)
        viewPager.adapter = SliderProfileAdapter(profileUiBinder, this, userList, rotation)
        viewPager.setCurrentItem(position.coerceIn(0, (userList.size - 1).coerceAtLeast(0)), false)
    }

    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
        val actionSilent = menu.findItem(R.id.action_notifications_off)
        val actionNotif = menu.findItem(R.id.action_notifications_on)
        val actionBloq = menu.findItem(R.id.action_bloq)
        val actionDesbloq = menu.findItem(R.id.action_desbloq)
        val actionDelete = menu.findItem(R.id.action_delete_chat)

        actionDelete.isVisible = true

        val currentIdx = viewPager.currentItem
        val current = userList.getOrNull(currentIdx)
        if (current != null) {
            firebaseRefsContainer.refDatos.child(user.uid).child(Constants.NODE_CURRENT_CHAT).child(current.id)
                .addValueEventListener(object : ValueEventListener {
                    override fun onDataChange(ds: DataSnapshot) {
                        val state = ds.child("estado").getValue(String::class.java)
                        when (state) {
                            "silent" -> {
                                actionSilent.isVisible = false
                                actionNotif.isVisible = true
                                actionDesbloq.isVisible = false
                                actionBloq.isVisible = true
                            }
                            Constants.NODE_CURRENT_CHAT, "delete", null -> {
                                actionSilent.isVisible = true
                                actionNotif.isVisible = false
                                actionDesbloq.isVisible = false
                                actionBloq.isVisible = true
                            }
                            "bloq" -> {
                                actionSilent.isVisible = false
                                actionNotif.isVisible = false
                                actionDesbloq.isVisible = true
                                actionBloq.isVisible = false
                            }
                        }
                    }
                    override fun onCancelled(error: DatabaseError) {}
                })
        }
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val u = userList.getOrNull(viewPager.currentItem) ?: return super.onOptionsItemSelected(item)
        val root = findViewById<android.view.View>(android.R.id.content)
        when (item.itemId) {
//            android.R.id.home -> onBackPressedDispatcher.onBackPressed()
            R.id.action_notifications_off -> {
                UserRepository.silent(u.name, u.id, Constants.CHATWITH)
                Toast.makeText(this, "Notificaciones desactivadas", Toast.LENGTH_SHORT).show()
            }
            R.id.action_notifications_on -> {
                UserRepository.silent(u.name, u.id, Constants.CHATWITH)
                Toast.makeText(this, "Notificaciones activadas", Toast.LENGTH_SHORT).show()
            }
            R.id.action_bloq -> UserRepository.setBlockUser(this, u.name, u.id, root, Constants.CHATWITH)
            R.id.action_desbloq -> UserRepository.setUnBlockUser(this, u.id, u.name, root, Constants.CHATWITH)
            R.id.action_delete_chat -> ChatUtils.deleteChat(this, u.id, u.name, root, Constants.CHATWITH)
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onPause() {
        super.onPause()
        lifecycleScope.launch { userRepository.setUserLastSeen() }
    }

    override fun onResume() {
        super.onResume()
        lifecycleScope.launch { userRepository.setUserOnline() }
    }
}
